spyder/plugins/pylint/widgets/pylintgui.py

```diff
@@ -310,14 +337,29 @@ def analyze_button_handler(self):
         """Try to start code analysis when Analyze button pressed."""
         self.start_analysis.emit()
 
+    def get_pylintrc_path(self, filename):
+        """Get the path to the most proximate pylintrc config to the file."""
+        parent = self.parentWidget()
+        if parent is not None:
+            project_dir = parent.main.projects.get_active_project_path()
+        else:
+            project_dir = None
+        search_paths = [
+            osp.dirname(filename),  # File's directory
+            getcwd_or_home(),  # Working directory
+            project_dir,  # Project directory
+            osp.expanduser("~"),  # Home directory

```

Comment:
I think we have a more robust func to do this on https://github.com/spyder-ide/spyder/blob/master/spyder/config/base.py#L153

Reply:
See previous comment for why we can't and shouldn't use that.